### PR TITLE
MC-173: Merge branch '1.1' into MC-173

### PR DESCRIPTION
**Pull request by damien-carcel**
| Q                    | A
| -------------------- | ---
| Bug fix?             | Yes
| New feature?         | Yes
| BC breaks?           | No
| Tests pass?          | Yes
| Checkstyle issues?*  | No
| PMD issues?**        | Yes
| Changelog updated?   | Yes
| Fixed tickets        | MC-173
| DB schema updated?   | No
| Migration script?    | No
| Doc PR               | No

### CODE REVIEW

File: Cleaner/ProductCleaner.php [hunk 1]
```diff
@@ -81,6 +97,26 @@ public function setNotCompleteAnymoreAction($notCompleteAnymoreAction)
...
+    {
+        return $this->removeProductsNotHandledByPim;
+    }
+
+    /**
+     * @param boolean $removeProductsNotHandledByPim
+     *
+     * @return ProductCleaner
+     */
+    public function setRemoveProductsNotHandledByPim($removeProductsNotHandledByPim)
```
Comment (fitn) [inline Cleaner/ProductCleaner.php]: Is there any way to cut these long names? Property should not take more than 30 chars.

Comment (damien-carcel) [inline Cleaner/ProductCleaner.php]: I intended to do a full code sniffer and mess detector check on code, cause there is so messy stuff like this all around.

File: Normalizer/CategoryNormalizer.php [hunk 2]
```diff
@@ -22,17 +23,23 @@ class CategoryNormalizer extends AbstractNormalizer
      */
     protected $categoryMappingManager;
 
+    /** @var   */
```
Comment (fitn) [inline Normalizer/CategoryNormalizer.php]: /** @var CategoryRepository */

File: Cleaner/ProductCleaner.php [hunk 3]
```diff
@@ -30,6 +30,25 @@ class ProductCleaner extends Cleaner
     protected $channel;
 
     /**
+     * @var array
+     */
```
Comment (willy-ahva) [inline Cleaner/ProductCleaner.php]: inline doc pls

File: Cleaner/ProductCleaner.php [hunk 4]
```diff
@@ -30,6 +30,25 @@ class ProductCleaner extends Cleaner
...
+    protected $productTypesNotHandledByPim = [
+        AbstractNormalizer::MAGENTO_BUNDLE_PRODUCT_KEY,
+        AbstractNormalizer::MAGENTO_DOWNLOADABLE_PRODUCT_KEY,
+        AbstractNormalizer::MAGENTO_VIRTUAL_PRODUCT_KEY,
+    ];
+
+    /**
+     * @var string
+     */
```
Comment (willy-ahva) [inline Cleaner/ProductCleaner.php]: inline doc pls

File: Cleaner/ProductCleaner.php [hunk 5]
```diff
@@ -30,6 +30,25 @@ class ProductCleaner extends Cleaner
...
+    protected $notCompleteAnymoreAction;
+
+    /**
+     * @var boolean
+     */
```
Comment (willy-ahva) [inline Cleaner/ProductCleaner.php]: inline doc pls

File: Cleaner/ProductCleaner.php [hunk 6]
```diff
@@ -188,20 +231,35 @@ protected function handleProductNotInPimAnymore(array $product)
...
-        $this->handleProduct($product, $this->notCompleteAnymoreAction);
+        $this->handleProduct(
+            $product,
+            $this->notCompleteAnymoreAction,
+            $this->removeProductsNotHandledByPim
+        );
     }
 
     /**
      * Handle product for the given action
```
Comment (willy-ahva) [inline Cleaner/ProductCleaner.php]: missing blank line

**Pull request merged**